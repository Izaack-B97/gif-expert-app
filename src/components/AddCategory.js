import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        // console.log(e.target.value);
        setInputValue(e.target.value);
    };
    
    const handleSubmit = (e) => {
        e.preventDefault(); // Cancela el comportamiento por default
        
        if(inputValue.trim().length > 2) {
            setCategories(categories => [ inputValue, ...categories ]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={ handleSubmit }>
            <h2>AddCategory</h2>
            <input type="text" value={ inputValue } onChange={ handleInputChange }/>
        </form>
    )
}

// Props requeridas
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
