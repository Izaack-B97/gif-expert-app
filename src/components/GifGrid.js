import React, { Fragment } from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
        
    const { data: images ,loading } = useFetchGifs(category);
    // console.log(loading);

    return (
        <Fragment>
            <h3 className="animate__animated animate__fadeIn">{ category }</h3>
            { loading && <p className="animate__animated animate__flash">Loading . . .</p> }
            <div className="card-grid">
                {
                    images.map(image => {
                        return <GifGridItem key={ image.id }{ ...image }/>
                    })
                }
            </div>
        </Fragment>
    )
}

GifGrid.propType = {
    category: PropTypes.string.isRequired
}