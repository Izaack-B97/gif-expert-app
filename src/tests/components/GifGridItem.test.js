import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem /> ' , () => {
    
    test('debe de mostrar correctamente el componente ', () => {

        const title = 'Mi titulo';
        const url = 'https://example.com';

        const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);
        
        expect( wrapper ).toMatchSnapshot();

    });
    
});