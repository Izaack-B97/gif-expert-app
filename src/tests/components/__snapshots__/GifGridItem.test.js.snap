// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem />  debe de mostrar correctamente el componente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Mi titulo"
    src="https://example.com"
  />
  <p>
    Mi titulo
  </p>
</div>
`;
